common/log/zap_logger.go

```diff
@@ -239,10 +241,45 @@ func buildZapLogger(cfg Config, disableCaller bool) *zap.Logger {
 		ErrorOutputPaths: []string{outputPath},
 		DisableCaller:    disableCaller,
 	}
-	logger, _ := config.Build()
+	var logger *zap.Logger
+	if cfg.EnableRotation && len(cfg.OutputFile) > 0 {
+		core, _ := rotationZapCore(cfg, config)
+		logger = zap.New(core)
+	} else {
+		logger, _ = config.Build()
+	}
 	return logger
 }
 
+func rotationZapCore(cfg Config, zapCfg zap.Config) (zapcore.Core, error) {
+	w := zapcore.AddSync(&lumberjack.Logger{
+		Filename:   cfg.OutputFile,
+		MaxSize:    cfg.MaxSize, // megabytes
+		MaxBackups: cfg.MaxBackups,
+		MaxAge:     cfg.MaxAge, // days
+	})
+	encoder, err := newEncoder(zapCfg.Encoding, zapCfg.EncoderConfig)
+	if err != nil {
+		return nil, err
+	}
+	core := zapcore.NewCore(
+		encoder,
+		w,
+		zapCfg.Level,
+	)
+	return core, nil
+}
+
+func newEncoder(name string, encoderConfig zapcore.EncoderConfig) (zapcore.Encoder, error) {
+	if name == "console" {
```

Comment:
Would you mind replacing these with an enum containing something like `ConsoleEncoding` and `JSONEncoding` in the few places we use it?